syncer/syncer_test.go

```diff
@@ -997,18 +1019,15 @@ func (s *testSyncerSuite) TestCasuality(c *C) {
 	c.Assert(err, IsNil)
 
 	syncer.setupMockCheckpoint(c, dbConn, mock)
-
 	mock.ExpectBegin()
 	mock.ExpectExec(".*INSERT INTO .* VALUES.* ON DUPLICATE KEY UPDATE.*").WillReturnResult(sqlmock.NewResult(0, 1))
 	mock.ExpectCommit()
-	key, err = syncer.resolveCasuality([]string{"a", "b"})
+	key4, err := syncer.resolveCasuality(keys[3])
 	c.Assert(err, IsNil)
-	c.Assert(key, Equals, "a")
-
+	c.Assert(key4, Equals, key1) // key4 conflict with key1
```

Comment:
```suggestion
	c.Assert(key4, Equals, keys[3][0])
```

Reply:
the result of "key4 conflict with key1" is generating a flush job and reset, the the meaning of returned key has no relation with key1

Reply:
fixed in https://github.com/pingcap/dm/pull/2043/commits/417880a1ca81f3659999b9e5c8862360b0f43311